firewood/src/merkle.rs

```diff
@@ -338,28 +325,17 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
                 };
 
             // [parent] (-> [ExtNode]) -> [branch with v] -> [Leaf]
-            let mut children = [None; NBRANCH];
+            let mut children = [None; BranchNode::MAX_CHILDREN];
 
             children[idx] = leaf_address.into();
 
-            let branch_address = self
-                .put_node(Node::branch(BranchNode {
-                    children,
-                    value,
-                    children_encoded: Default::default(),
-                }))?
-                .as_ptr();
-
-            if !prefix.is_empty() {
-                self.put_node(Node::from(NodeType::Extension(ExtNode {
-                    path: PartialPath(prefix.to_vec()),
-                    child: branch_address,
-                    child_encoded: None,
-                })))?
-                .as_ptr()
-            } else {
-                branch_address
-            }
+            self.put_node(Node::from_branch(BranchNode {
```

Comment:
(comment only) this is fine, and quite different semantics from `from_leaf`, which doesn't require a `LeafNode`

Reply:
See comment above